### PR TITLE
77658 download registration submissions

### PR DESCRIPTION
**Pull request by will-hope98**

### CODE REVIEW

File: src/EA.Weee.Web/Views/Shared/Producer/ViewOrganisation/ContactDetails.cshtml [hunk 1]
```diff
@@ -80,6 +80,22 @@
                 </tbody>
             </table>
         </div>
+        @if (Model.IsInternalAdmin)
+        {
+            <div>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/Shared/Producer/ViewOrganisation/ContactDetails.cshtml:85]: @will-hope98  not sure how easy this is. But is it possible to put this into the _OrganisationDetailsTabs.cshtml to save having it in each view.

File: src/EA.Weee.Web/Views/Shared/_CheckAnswersContentPartial.cshtml [hunk 2]
```diff
@@ -163,12 +163,15 @@
...
-        <tr class="govuk-table__row">
-            <th scope="col" class="govuk-table__header govuk-!-text-align-left" style="width: 28%;">Selling technique(s) that you use</th>
-            <td class="govuk-table__cell" colspan="2">
-                @Model.EeeData.SellingTechnique.ToSellingTechniqueType().GetDisplayName()
-            </td>
-        </tr>
+        @if (@Model.EeeData.SellingTechnique.IsDirectSelling || Model.EeeData.SellingTechnique.IsIndirectSelling)
+        {
+            <tr class="govuk-table__row">
+                <th scope="col" class="govuk-table__header govuk-!-text-align-left" style="width: 28%;">Selling technique(s) that you use</th>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/Shared/_CheckAnswersContentPartial.cshtml:169]: @will-hope98 Just thinking about this I wonder if we should always display the row it should just not display the value if it doesnt have value.

File: src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/CheckAnswersMap.cs [hunk 3]
```diff
@@ -46,10 +47,17 @@ public CheckAnswersViewModel Map(SmallProducerSubmissionMapperData source)
...
+                ComplianceYear = year
             };
 
             return viewModel;
         }
+
+        public CheckAnswersViewModel Map(SmallProducerSubmissionMapperData source)
+        {
+            // Uses the current year if no explicit year is provided
+            int year = source.Year ?? DateTime.Now.Year;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/CheckAnswersMap.cs:59]: @will-hope98  think if the source year isn't set we could probably get away with using submissionData.CurrentSubmission.ComplianceYear

File: src/EA.Weee.Web/Views/Shared/_CheckAnswersContentPartial.cshtml (hunk 2 above)
Comment (will-hope98) [reply src/EA.Weee.Web/Views/Shared/_CheckAnswersContentPartial.cshtml:169]: @JohnPadgettCivica Would have to create a new partial I think. if you prefer to do it that way

Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Views/Shared/_CheckAnswersContentPartial.cshtml:169]: @will-hope98 Maybe move that into a partial and move the if statement to be inside the cell that renders the actual value so it displays nothing the row title but empty value when its not present.

File: src/EA.Weee.Web/Views/Shared/Producer/ViewOrganisation/ContactDetails.cshtml (hunk 1 above)
Comment (will-hope98) [reply src/EA.Weee.Web/Views/Shared/Producer/ViewOrganisation/ContactDetails.cshtml:85]: @JohnPadgettCivica Would have to create a new partial I think. if you prefer to do it that way

Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Views/Shared/Producer/ViewOrganisation/ContactDetails.cshtml:85]: @will-hope98 OK as this is at the bottom of the page. Stick it in a partial then as that seems to be best way of re-using

**Comment by JohnPadgettCivica** [comment]
Not sure if this is quite working correctly.

2025 Sub
image


2026 sub updates the contact details
image

When i download the 2025 PDF its showing the updated 2026 details
image

File: src/EA.Weee.Web/Areas/Producer/ViewModels/OrganisationDetailsTabsViewModel.cs [hunk 4]
```diff
@@ -29,7 +29,7 @@ public class OrganisationDetailsTabsViewModel
 
         public bool ShowReturnRegistrationToUser => this.IsInternalAdmin && (IsRegistered || this.Status == SubmissionStatus.Submitted);
 
-        public bool ShowContinueRegistrationToUser => !this.IsInternalAdmin && this.Status == SubmissionStatus.InComplete && this.CurrentYear == this.Year;
+        public bool ShowContinueRegistrationToUser => !this.IsInternalAdmin && this.Status != SubmissionStatus.Submitted && this.CurrentYear == this.Year;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Producer/ViewModels/OrganisationDetailsTabsViewModel.cs:32]: @will-hope98  update the unit tests that were initially created for this.

**Pull request merged**